test/conformance/ptp/ptp.go

```diff
@@ -1081,3 +1138,237 @@ func printInterface(config ptpv1.PtpConfig, interfaceDetailsMap map[string]strin
 		}
 	}
 }
+
+func rebootSlaveNode(fullConfig testconfig.TestConfig) {
+	logrus.Info("Rebooting system starts ..............")
+
+	const (
+		daemonSetName    = "ptp-reboot"
+		containerName    = "container-00"
+		imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+
+		timeout = 5 * time.Minute
+	)
+
+	// 1. create a daemon set for the node reboot
+	dsRunningPods, err := daemonsets.CreateDaemonSet(daemonSetName, defaultNamespace, containerName, imageWithVersion, timeout)
+	if err != nil {
+		logrus.Errorf("Error : +%v\n", err.Error())
+	}
+	nodeToPodMapping := make(map[string]v1core.Pod)
+	for _, dsPod := range dsRunningPods.Items {
+		nodeToPodMapping[dsPod.Spec.NodeName] = dsPod
+	}
+
+	// 2. Get the slave config, restart the slave node
+	slavePtpConfig := fullConfig.DiscoveredSlavePtpConfig.Config
+
+	restartedNodes := make([]string, len(dsRunningPods.Items))
+
+	for _, recommend := range slavePtpConfig.Spec.Recommend {
+		matches := recommend.Match
+
+		for _, match := range matches {
+			nodeLabel := *match.NodeLabel
+			// Get all nodes with this node label
+			nodes, err := client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: nodeLabel})
+			if err != nil {
+				panic(err.Error())
+			}
+			// Restart the node
+			for _, node := range nodes.Items {
+				rebootNode(nodeToPodMapping[node.Name], node)
+				restartedNodes = append(restartedNodes, node.Name)
+			}
+		}
+	}
+	logrus.Printf("Restarted nodes %v", restartedNodes)
+
+	// 3. Find the ptp in slave machine
+	ptpRunningPods := checkAndGetPtpRunningPods(fullConfig)
+
+	// 4. Verify the setup of PTP
+	verifyAfterRebootState(restartedNodes, fullConfig)
+
+	// 5. slave can sync to master
+	checkSlaveSyncWithMaster(ptpRunningPods, fullConfig)
+
+	logrus.Info("Rebooting system ends ..............")
+}
+
+func rebootNode(pod v1core.Pod, node v1core.Node) {
+	checkRestart(pod)
+	// Wait for the node to be unreachable
+	const unrechable = false
+	testnode.WaitForNodeReachability(&node, 5*time.Minute, unrechable)
+
+	// Wait for all nodes to be reachable now after the restart
+	const reachable = true
```

Comment:
try to use enum instead, 
type nodeStatus int
const (
	reachable nodeStatus = iota
        unreacble   
)

or have const clubbed together 
const (
reachable bool =true
unreacble bool= false
)

Reply:
WaitForNodeReachability the funx name does says what it is , so true or false value alters the behavior either waitForNodeReachbility  or not ?
Btw what is this trying to do ?
The un-reachable checks for  5 minutes  and  doesn't indicate if the node became un-reachable.
What if the node never restarted at this state and its still reachable?
Are you supposed to fail if the node never rebooted?

Reply:
WaitForNodeReachability is for waiting for the node reachability (reachable or unreachable) mentioned by the last argument. When you pass true, it means wait until the node is reachable. When you pass false, it means wait until the node is unreachable. Those variables are used for the readability of what this function is achieving for. 

In bare metal cluster, the timeout can be longer for the unreachability of the node, hence I plan to increase 5 mins to 10 mins. 

Are you supposed to fail if the node never rebooted?
- There is no way to check if the node is restarted within the timeout. Only way is to increase the timeout to very high and hope that the desired state is reached.